src/k8s/pkg/utils/k8s/client.go

```diff
@@ -23,3 +23,15 @@ func NewClient(snap snap.Snap) (*Client, error) {
 	}
 	return &Client{clientset}, nil
 }
+
+func NewNodeClient(snap snap.Snap) (*Client, error) {
+	config, err := snap.KubernetesNodeRESTClientGetter("kube-system").ToRESTConfig()
+	if err != nil {
+		return nil, fmt.Errorf("failed to build Kubernetes REST config: %w", err)
+	}
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create Kubernetes clientset: %w", err)
+	}
+	return &Client{clientset}, nil
+}
```

Comment:
Rather, make this a single `NewClient()` that accepts the restclientgetter and the namespace as arguments. The kubernetes client package does not _care_ about what kind of client it will create.

Reply:
I've updated it, not exactly sure if this was what you envisioned. Let me know if this needs more changing, thank you!